From b4ef61c0dc1ee59830f05136210214262a5f47f9 Mon Sep 17 00:00:00 2001
From: Jason Rogena <j.rogena@cgiar.org>
Date: Thu, 29 May 2014 17:41:28 +0300
Subject: [PATCH 178/213] Version 0.9.8.1: Date popping up in AddEventActivity,
 Calving event in EventHistoryActivity

The date dialog now doesn't popup in AddEventActivity

Calving event in EventHistoryActivity now Showing up in list
---
 MistroFarmer/src/main/AndroidManifest.xml                             | 2 +-
 .../src/main/java/org/cgiar/ilri/mistro/farmer/AddEventActivity.java  | 3 +++
 .../main/java/org/cgiar/ilri/mistro/farmer/EventsHistoryActivity.java | 3 +++
 MistroFarmer/src/main/res/layout/activity_add_event.xml               | 4 ++++
 4 files changed, 11 insertions(+), 1 deletion(-)

diff --git a/MistroFarmer/src/main/AndroidManifest.xml b/MistroFarmer/src/main/AndroidManifest.xml
index a29a593..18fbbcb 100644
--- a/MistroFarmer/src/main/AndroidManifest.xml
+++ b/MistroFarmer/src/main/AndroidManifest.xml
@@ -2,7 +2,7 @@
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
     package="org.cgiar.ilri.mistro.farmer"
     android:versionCode="8"
-    android:versionName="0.9.7.1" >
+    android:versionName="0.9.8.1" >
 
     <uses-sdk
         android:minSdkVersion="7"
diff --git a/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/AddEventActivity.java b/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/AddEventActivity.java
index 3b2e1c7..95f73f0 100644
--- a/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/AddEventActivity.java
+++ b/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/AddEventActivity.java
@@ -58,6 +58,7 @@ public class AddEventActivity extends SherlockActivity implements MistroActivity
     public static final String MODE_CALVING="Birth";
     public static final String KEY_SERVICING_TYPE="servicingType";
 
+    private EditText phantomET;
     private Menu menu;
     private TextView cowIdentifierTV;
     private Spinner cowIdentifierS;
@@ -118,6 +119,8 @@ public class AddEventActivity extends SherlockActivity implements MistroActivity
 
         cacheData = true;
 
+        phantomET = (EditText) this.findViewById(R.id.phantom_et);
+        phantomET.requestFocus();
         cowIdentifierTV=(TextView)findViewById(R.id.cow_identifier_tv);
         cowIdentifierS=(Spinner)findViewById(R.id.cow_identifier_s);
         dateTV=(TextView)findViewById(R.id.date_tv);
diff --git a/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/EventsHistoryActivity.java b/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/EventsHistoryActivity.java
index 6c602b4..d6ad9bf 100644
--- a/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/EventsHistoryActivity.java
+++ b/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/EventsHistoryActivity.java
@@ -319,6 +319,9 @@ public class EventsHistoryActivity extends SherlockActivity implements MistroAct
                     if(eventTypesInEN[j].equals(jsonObject.getString("event_name"))) {
                         index = j;
                     }
+                    else if(jsonObject.getString("event_name").equals("Calving")){
+                        eventTypeString = Locale.getStringInLocale("calving", EventsHistoryActivity.this);
+                    }
                 }
                 if(eventTypesInEN.length == eventTypes.length && index != -1) {
                     eventTypeString = eventTypes[index];
diff --git a/MistroFarmer/src/main/res/layout/activity_add_event.xml b/MistroFarmer/src/main/res/layout/activity_add_event.xml
index 4681c67..19a1194 100644
--- a/MistroFarmer/src/main/res/layout/activity_add_event.xml
+++ b/MistroFarmer/src/main/res/layout/activity_add_event.xml
@@ -13,6 +13,10 @@
         android:paddingLeft="@dimen/activity_padding_side"
         android:paddingRight="@dimen/activity_padding_side"
         android:orientation="vertical">
+        <EditText
+            android:id="@+id/phantom_et"
+            android:layout_width="match_parent"
+            android:layout_height="1dp"/>
         <TextView
             android:id="@+id/event_type_tv"
             android:layout_height="wrap_content"
-- 
2.7.1

